nixos/tests/boot.nix

```diff
@@ -44,5 +44,41 @@ in {
         usb => glob("${iso}/iso/*.iso"),
         bios => '${pkgs.OVMF}/FV/OVMF.fd'
       '';
-  }
-
+    netboot = let
+      config = (import ../lib/eval-config.nix {
+          inherit system;
+          modules =
+            [ ../modules/installer/netboot/netboot.nix
+              ../modules/testing/test-instrumentation.nix
+              { key = "serial"; }
+            ];
+        }).config;
+      ipxeScriptDir = pkgs.writeTextFile {
+        name = "ipxeScriptDir";
+        text = ''
+          #!ipxe
+          dhcp
+          kernel bzImage init=${config.system.build.toplevel}/init ${toString config.boot.kernelParams} console=ttyS0
```

Comment:
I *think* it would be better to leave out `console=ttyS0` here, since it will not be useful on physical systems and might cause confusion in combination with `config.boot.kernelParams`.

It does mean the test script will not work, however with the automated netboot test Nix would add console anyway.